pkg/agent/flowexporter/connections/l7_listener_test.go

```diff
@@ -0,0 +1,312 @@
+						Method:        "GET",
+						Protocol:      "HTTP/1.1",
+						Status:        200,
+						ContentLength: 153,
+					},
+				},
+			},
+			eventPresent: true,
+			expectedEvents: L7ProtocolFields{
+				http: []map[int32]*Http{
+					{
+						0: &Http{
+							Hostname:      "10.10.0.1",
+							URL:           "/public/1",
+							UserAgent:     "curl/7.74.0",
+							ContentType:   "text/html",
+							Method:        "GET",
+							Protocol:      "HTTP/1.1",
+							Status:        200,
+							ContentLength: 153,
+						},
+					},
+				},
+			},
+		}, {
+			name: "Valid case for persistent http",
+			input: []JsonToEvent{
+				{
+					Timestamp:   time.Now().String(),
+					FlowID:      1,
+					InInterface: "mock_interface",
+					EventType:   "http",
+					VLAN:        []int32{1},
+					SrcIP:       netip.MustParseAddr("10.10.0.1"),
+					SrcPort:     59920,
+					DestIP:      netip.MustParseAddr("10.10.0.2"),
+					DestPort:    80,
+					Proto:       "TCP",
+					TxID:        0,
+					HTTP: &Http{
+						Hostname:      "10.10.0.1",
+						URL:           "/public/2",
+						UserAgent:     "curl/7.74.0",
+						ContentType:   "text/html",
+						Method:        "GET",
+						Protocol:      "HTTP/1.1",
+						Status:        200,
+						ContentLength: 153,
+					},
+				}, {
+					Timestamp:   time.Now().String(),
+					FlowID:      1,
+					InInterface: "mock_interface",
+					EventType:   "http",
+					VLAN:        []int32{1},
+					SrcIP:       netip.MustParseAddr("10.10.0.1"),
+					SrcPort:     59920,
+					DestIP:      netip.MustParseAddr("10.10.0.2"),
+					DestPort:    80,
+					Proto:       "TCP",
+					TxID:        1,
+					HTTP: &Http{
+						Hostname:      "10.10.0.1",
+						URL:           "/public/3",
+						UserAgent:     "curl/7.74.0",
+						ContentType:   "text/html",
+						Method:        "GET",
+						Protocol:      "HTTP/1.1",
+						Status:        201,
+						ContentLength: 154,
+					},
+				},
+			},
+			eventPresent: true,
+			expectedEvents: L7ProtocolFields{
+				http: []map[int32]*Http{
+					{
+						0: &Http{
+							Hostname:      "10.10.0.1",
+							URL:           "/public/2",
+							UserAgent:     "curl/7.74.0",
+							ContentType:   "text/html",
+							Method:        "GET",
+							Protocol:      "HTTP/1.1",
+							Status:        200,
+							ContentLength: 153,
+						},
+					}, {
+						1: &Http{
+							Hostname:      "10.10.0.1",
+							URL:           "/public/3",
+							UserAgent:     "curl/7.74.0",
+							ContentType:   "text/html",
+							Method:        "GET",
+							Protocol:      "HTTP/1.1",
+							Status:        201,
+							ContentLength: 154,
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			socketConn, err := net.Dial("unix", l.suricataEventSocketPath)
+			if err != nil {
+				t.Fatalf("Failed to connect to server: %s", err)
+			}
+			defer socketConn.Close()
+			if tc.name != "Invalid eventType" {
```

Comment:
https://github.com/antrea-io/antrea/pull/5218#discussion_r1410935522 not addressed, test code shouldn't rely on test name.